hack/update/inspektor_gadget_version/update_inspektor_gadget_version.go

```diff
@@ -58,4 +62,42 @@ func main() {
 	klog.Infof("inspektor-gadget stable version: %s", data.Version)
 
 	update.Apply(schema, data)
+	updateDeploymentYAML(stable)
+	updateCRDYAML(stable)
+}
+
+func updateDeploymentYAML(version string) {
+	res, err := http.Get(fmt.Sprintf("https://raw.githubusercontent.com/inspektor-gadget/inspektor-gadget/refs/tags/%s/pkg/resources/manifests/deploy.yaml", version))
+	if err != nil {
+		klog.Fatalf("failed to get yaml file: %v", err)
+	}
+	defer res.Body.Close()
+	yaml, err := io.ReadAll(res.Body)
+	if err != nil {
+		klog.Fatalf("failed to read body: %v", err)
+	}
+	replacements := map[string]string{
+		`ghcr\.io\/inspektor-gadget\/inspektor-gadget:.*`: "{{.CustomRegistries.InspektorGadget  | default .ImageRepository | default .Registries.InspektorGadget }}{{.Images.InspektorGadget}}",
```

Comment:
Thanks for pointing this out, the problem is that `{{.Images.InspektorGadget}}` includes the image and the tag, where there seems to be a specific env for the image and tag individually. This in itself if not a big issue because I could make two new map fields `InspektorGadgetWithoutVersion` and `InspektorGadgetVersion` (or something similar). But this is further complicated that we allow the user to overwrite the `.Images.InspektorGadget` using flags. So then the envs and the image actually used would be out of sync again. Just curious, what are the `GADGET_IMAGE` & `INSPEKTOR_GADGET_VERSION` envs used for.